### PR TITLE
Another BugFix for 4.24 and some implementation adjust

### PR DESCRIPTION
**Pull request by dorgonman**
This PR fix many bugs for me in 4.24

I also notice it will crash the game when bStoreGameInstance is true, not sure why, so I set it to false by default.

### CODE REVIEW

File: Source/SaveExtension/Private/SaveManager.cpp [hunk 1]
```diff
@@ -110,11 +110,11 @@ bool USaveManager::DeleteSlot(int32 SlotId)
 	bool bSuccess = false;
 
 	MTTasks.CreateTask<FDeleteSlotsTask>(this, SlotId)
-	.OnFinished([bSuccess](auto& Task) mutable {
+	.OnFinished([&bSuccess](auto& Task) mutable {
```
Comment (muit) [inline Source/SaveExtension/Private/SaveManager.cpp:113]: Nice catch!

File: Source/SaveExtension/Private/Serialization/SlotDataTask_Saver.cpp [hunk 2]
```diff
@@ -117,7 +117,19 @@ void USlotDataTask_Saver::Tick(float DeltaTime)
 	if (SaveInfoTask && SaveDataTask && 
 		SaveInfoTask->IsDone() && SaveDataTask->IsDone())
 	{
-		Finish(true);
+		if (bSaveThumbnail)
+		{
+			USaveManager* Manager = GetManager();
+			USlotInfo* CurrentInfo = Manager->GetCurrentInfo();
+			if (CurrentInfo->GetThumbnail())
```
Comment (muit) [inline Source/SaveExtension/Private/Serialization/SlotDataTask_Saver.cpp:124]: This is interesting. So you wait for the screenshot to be made, good idea.

Doesn't the SlotDataTask already have a pointer to SlotInfo?
You may not need to find it using the Manager.

File: Source/SaveExtension/Public/SaveManager.h [hunk 3]
```diff
@@ -297,9 +298,14 @@ class SAVEEXTENSION_API USaveManager : public UGameInstanceSubsystem, public FTi
...
-			return PresetAsset.LoadSynchronous();
+			USavePreset* SavePreset = PresetAsset.LoadSynchronous();
+			ensureMsgf(SavePreset, TEXT("Can't load %s"), *PresetAsset.GetLongPackageName());
+			if (SavePreset)
+			{
+				return SavePreset;
+			}
 		}
-		return GetDefault<USavePreset>();
+		return GetDefault<USavePreset>();;
```
Comment (muit) [inline Source/SaveExtension/Public/SaveManager.h:308]: Misspelling

File: Source/SaveExtension/Private/Serialization/SlotDataTask_Saver.cpp (hunk 2 above)
Comment (dorgonman) [reply Source/SaveExtension/Private/Serialization/SlotDataTask_Saver.cpp]: done

Comment (dorgonman) [reply Source/SaveExtension/Private/Serialization/SlotDataTask_Saver.cpp]: done

File: Source/SaveExtension/Public/SaveManager.h (hunk 3 above)
Comment (dorgonman) [reply Source/SaveExtension/Public/SaveManager.h]: done

**Comment by muit** [comment]
Thanks for the contribution :)

**Pull request merged**